File: src/main/java/com/plugin/frege/editor/FregeCommenter.java
package com.plugin.frege.editor;

import com.intellij.lang.Commenter;

public class FregeCommenter implements Commenter {

    @Override
    public String getLineCommentPrefix() {
        return "--";
    }

    @Override
    public String getBlockCommentPrefix() {
        return "{--";
    }

    @Override
    public String getBlockCommentSuffix() {
        return "-}";
    }

    @Override
    public String getCommentedBlockCommentPrefix() {
        return "{--";
    }

    @Override
    public String getCommentedBlockCommentSuffix() {
        return "-}";
    }
}

File: src/main/kotlin/com/plugin/frege/editor/FregeCommenter.kt
package com.plugin.frege.editor

import com.intellij.lang.Commenter

class FregeCommenter : Commenter {
    override fun getLineCommentPrefix(): String = "--"

    override fun getBlockCommentPrefix(): String = "{--"

    override fun getBlockCommentSuffix(): String = "-}"

    override fun getCommentedBlockCommentPrefix(): String = "{--"

    override fun getCommentedBlockCommentSuffix(): String = "-}"
}
